airflow/migrations/env.py

```diff
@@ -51,7 +54,20 @@ def include_object(_, name, type_, *args):
 # my_important_option = config.get_main_option("my_important_option")
 # ... etc.
 
-COMPARE_TYPE = False
+
+def compare_type(context, inspected_column, metadata_column, inspected_type, metadata_type):
+    # return False if the metadata_type is the same as the inspected_type
+    # or None to allow the default implementation to compare these
+    # types. a return value of True means the two types do not
+    # match and should result in a type change operation.
+    if isinstance(inspected_type, DATETIME) and isinstance(metadata_type, UtcDateTime):
+        # for sqlite
+        return False
+    if isinstance(inspected_type, mysql.VARCHAR) and isinstance(metadata_type, String):
+        # This is a hack to get around MySQL VARCHAR collation
+        # not being possible to change from utf8_bin to utf8mb3_bin
+        return False
```

Comment:
Would it be possible to do these only on corresponding databases? I _think_ the info is available on `context`.